client/components/import/import.js

```diff
@@ -88,3 +145,110 @@ ImportPopup.extendComponent({
+  members() {
+    return Session.get('import.membersToMap');
+  },
+  _refreshMembers(listOfMembers) {
+    Session.set('import.membersToMap', listOfMembers);
+  },
+  /**
+   * Will look into the list of members to import for the specified memberId,
+   * then set its property to the supplied value.
+   * If unset is true, it will remove the property from the rest of the list as well.
+   *
+   * use:
+   * - memberId = null to use selected member
+   * - value = null to unset a property
+   * - unset = true to ensure property is only set on 1 member at a time
+   */
+  _setPropertyForMember(property, value, memberId, unset = false) {
+    const listOfMembers = this.members();
+    let finder = null;
+    if(memberId) {
+      finder = (member) => member.id === memberId;
+    } else {
+      finder = (member) => member.selected;
+    }
+    listOfMembers.forEach((member) => {
+      if(finder(member)) {
+        if(value !== null) {
+          member[property] = value;
+        } else {
+          delete member[property];
+        }
+        if(!unset) {
+          // we shortcut if we don't care about unsetting the others
+          return false;
+        }
+      } else if(unset) {
+        delete member[property];
+      }
+      return true;
+    });
+    // Session.get gives us a copy, we have to set it back so it sticks
+    this._refreshMembers(listOfMembers);
+  },
+  setSelectedMember(memberId) {
+    return this._setPropertyForMember('selected', true, memberId, true);
+  },
+  /**
+   * returns the member with specified id,
+   * or the selected member if memberId is not specified
+   */
+  getMember(memberId = null) {
+    const allMembers = Session.get('import.membersToMap');
+    let finder = null;
+    if(memberId) {
+      finder = (user) => user.id === memberId;
+    } else {
+      finder = (user) => user.selected;
+    }
+    return allMembers.find(finder);
+  },
+  mapSelectedMember(wekan) {
+    return this._setPropertyForMember('wekan', wekan, null);
+  },
+  unmapMember(memberId){
+    return this._setPropertyForMember('wekan', null, memberId);
+  },
+});
+
+ImportMapMembers.extendComponent({
+  onMapMember(evt) {
+    const memberToMap = this.currentData();
+    if(memberToMap.wekan) {
+      // todo xxx ask for confirmation?
+      this.unmapMember(memberToMap.id);
+    } else {
+      this.setSelectedMember(memberToMap.id);
+      Popup.open('mapMembersAdd')(evt);
+    }
+  },
+  onSubmit(evt) {
+    evt.preventDefault();
+    Popup.back();
+  },
+  events() {
+    return [{
+      'submit': this.onSubmit,
+      'click .mapping': this.onMapMember,
+    }];
+  },
+}).register('mapMembersPopup');
+
+ImportMapMembers.extendComponent({
+  onSelectUser(){
+    this.mapSelectedMember(this.currentData());
+    Popup.back();
+  },
+  events() {
+    return [{
+      'click .js-select-import': this.onSelectUser,
+    }];
+  },
+  onRendered() {
+    // todo XXX why do I not get the focus??
```

Comment:
I will look at it, but I had some similar issues in the past. It’s probably an issue with reactivity (ie the `input` is redrawn after the `focus()` call, or something like that).